org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/environments/EnvironmentsManager.java

```diff
@@ -253,6 +253,16 @@ private String getExecutionEnvironmentCompliance(IExecutionEnvironment execution
 			return JavaCore.VERSION_1_1;
 		} else if (desc.indexOf("1.0") != -1) { //$NON-NLS-1$
 			return "1.0"; //$NON-NLS-1$
+		} else {
+			// an optimistic strategy for not-yet-declared versions
```

Comment:
That produces `1` from `25.0.1`, and that is really **guessing**...
I'm not sure this is the right way to guess the version out of EE.
The entire method looks like a big hack.
Why can't we add an extra "JavaCompliance" attribute to `org.eclipse.jdt.launching.executionEnvironments` extension point and specify there all what we need?

Reply:
> That produces 1 from 25.0.1, and that is really guessing..

The current code would also return 24 for 1.0.24, so it's not so much worse...

> Why can't we add an extra "JavaCompliance" attribute to org.eclipse.jdt.launching.executionEnvironments extension point and specify there all what we need?

We sure can, but usually, this happens on a branch that takes ages before it is merged...